tuple-table/src/main/java/com/iluwatar/tupletable/MemberTupleDao.java

```diff
@@ -0,0 +1,149 @@
+package com.iluwatar.tupletable;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Types;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class MemberTupleDao {
+  /**
+   * Method implemented to find the member based on the input param.
+   *
+   * @param memberNo is key with which tuple record is looked up
+   * @return member retrieved from the database
+   */
+  public MemberDto findMember(long memberNo) throws SQLException, ClassNotFoundException {
+    Database db = new Database();
+    MemberDto member = new MemberDto();
+    member.setMemberNumber(memberNo);
+
+    try (Connection con = db.getConnection();
+         PreparedStatement ps = con.prepareStatement("select fieldname, numerical, string "
+                 + "from object_data where obj_pk = ?")) {
+      // create a new table
+      ps.setLong(1, memberNo);
+      try (ResultSet rs = ps.executeQuery()) {
+        while (rs.next()) {
+          String fieldName = rs.getString(1);
+          String strVal = rs.getString(3);
+          if (strVal != null) {
+            setVal(fieldName, member, strVal);
+          } else {
+            //we do this indirectly to make database typecasting more reliable
+            long lngVal = rs.getLong(2);
+            if (!rs.wasNull()) {
+              setVal(fieldName, member, lngVal);
+            }
+          }
+        }
+      }
+    }
+    return member;
+  }
+
+  /**
+   * Method to create the table if not exist (for demo purpose).
+   *
+   * @should create the object_data table when called
+   */
+  public void createTableIfNotExists() throws SQLException, ClassNotFoundException {
+    String sql = "CREATE TABLE IF NOT EXISTS object_data (\n"
+            + "OBJ_PK int NOT NULL,\n"
+            + "FIELDNAME varchar(20) NOT NULL,\n"
+            + "NUMERICAL int,\n"
+            + "STRING varchar(255)\n"
+            + ");";
+    Database db = new Database();
```

Comment:
Database object probably shouldn't be created separately in every method

Reply:
done, moved as class variable